src/main/java/com/stripe/model/Invoice.java

```diff
@@ -404,4 +406,20 @@ public Invoice voidInvoice(Map<String, Object> params, RequestOptions options)
         instanceUrl(Invoice.class, this.getId())), params, Invoice.class, options);
   }
   // </editor-fold>
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class ThresholdReason extends StripeObject {
+    Long amountGte;
+    List<ItemReason> itemReasons;
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class ItemReason extends StripeObject {
+      List<String> lineItemIds;
+      Long usageGte;
```

Comment:
Should the innermost class, when referred outside this class (in normal usage) will be
`Invoice.ThresholdItemReason` instead of the 3-level nesting as `Invoice.ThresholdReason.ItemReason` here? I think the former reads better.
So we can implement the following:
```
public static class ThresholdReason extends StripeObject {
  List<ThresholdItemReason> itemReasons
}
// the inner-most class is now at nested in Invoice instead
public static class ThresholdItemReason extends StripeObject {
}
```
Actually, in the autogen SDK, we currently enforce nesting at most two levels to flatten classes. It help to flatten long resource dependency we have in pay-server annotation too.
The `Charge.Level3.LineItem` will be in autogen (hypothetically) as `Charge.Level3LineItem` (although we decided on unwinding it to top level resource)
```
public class Charge {
  public static class Level3 extends StripeObject {
    List<Level3LineItem> lineItems;
    ....
  }
  public static class Level3LineItem extends StripeObject {
     ....
  }
}
```

Reply:
Fixed!